src/quo/components/tags/collectible_tag/view.cljs

```diff
@@ -0,0 +1,61 @@
+(ns quo.components.tags.collectible-tag.view
+  (:require
+    [oops.core :refer [oget]]
+    [quo.components.icon :as icons]
+    [quo.components.markdown.text :as text]
+    [quo.components.tags.collectible-tag.style :as style]
+    [quo.theme :as quo.theme]
+    [react-native.core :as rn]
+    [react-native.hole-view :as hole-view]
+    [reagent.core :as reagent]))
+
+(defn- view-internal
+  "Options:
+    - :options - false / :add / :hold (default false)
+    - :size - :size-24 / :size-32 (default :size-24)
+    - :blur? - boolean (default false)
+    - :theme - :light / :dark
+    - :collectible-name - string
+    - :collectible-number - string"
+  []
+  (let [container-width (reagent/atom 0)]
+    (fn [{:keys [options size blur? theme collectible-img-src collectible-name collectible-number]
+          :or   {size :size-24}}]
+      [rn/view
+       {:on-layout #(reset! container-width
```

Comment:
@J-Son89 @FFFra Am fixing a schema issue from this PR, but saw the layout and width being passed as props and am wondering why it's better this way? 

It seems like the width is needed for the component internals to know where to position the gap for the icon thingie in the corner. I don't think that's something the consumer of the component should be concerned with and can be done inside the component itself.

Reply:
yeah perhaps we need to revisit this. Actually the only other place I have seen the component handling the on-layout internally is the slider button and that component has a lot of issues when it needs to rerender as often the size and some other styles just persist from the initial render. I'm not exactly sure it's the on-layout causing this as there's some other initialisation steps there too. Probably this component is less likely to have issues like that as we will have little uses for this where as the slider button is in many screens used in different ways.

Reply:
yeah, for the slide-button the initial assumption is that it won't change its size/dimensions, as it might interfere with animations. Is this happening on a screen somewhere or in just the preview?

Reply:
I'm in the middle of writing some integration tests. Once I finish that I'll reach out to you and we can investigate it a little as there some examples we need to fix with it, I can show you! 👌